### ISSUE TITLE
“LightGBMError: Bug. There should be only one multi-val group.”

### ISSUE THREAD

**Issue by stakas** [microsoft/LightGBM#3458]
Apologies for turning here in a StackOverflow fashion, but I have not found any other kind of LightGBM help, so this is the last resort.

I am loading a series of large binary independently binarized with `save_binary` LightGBM files and concatenating them into a single file to train GBT with LightGBM, however, `train` method fails with a cryptic error message:

>"LightGBMError: Bug. There should be only one multi-val group."

I haven't found anything anywhere regarding this error and had a peek into C++ code and it seems that there is a fix pending regarding this:

https://github.com/microsoft/LightGBM/blob/0c1c36cdbdf771a03c9ae0873eb28585e6c3d624/src/io/dataset.cpp#L1502

In the meantime, can somebody advise me how to change my data so that the Dataset contruction would work? Thank you in advance. 

My snippet is as follows:

```
import lightgbm as lgb

my_files = ['p1.bin', 'p2.bin', 'p3.bin']

full_data = lgb.Dataset(my_files[0]).construct()

for f in my_files[1:]:        
    full_data.add_features_from(lgb.Dataset(f).construct())  

feature_name = ['feature_' + str(col) for col in range(full_data.num_feature())]
full_data.feature_name = feature_name

params = {'objective': 'binary', 
         'learning_rate': 0.01,
         'metric': ['binary'],
        }

bst = lgb.train(params = params, 
                train_set = full_data, 
                num_boost_round = 500,
                feature_name = feature_name,
                ) 
```

**Comment by StrikerRUS** [comment]
Hello @stakas !

Could you please try this branch?
https://github.com/microsoft/LightGBM/tree/fix-add-features

I hope #2754 will fix this issue.

**Comment by stakas** [comment]
Ah, I had a hunch that is related to the addition of data sets, just could not get through the woods of C code. Will get some sleep and try `fix-add-features` branch.

Your help much is appreciated @StrikerRUS !

**Comment by StrikerRUS** [comment]
Fixed in #2754. Feel free to reopen if the error hasn't gone.
